File: 이예진/깊이, 너비 우선 탐색(DFS, BFS)/네트워크.py
def find(a):
    global par
    if par[a] != a:
        par[a] = find(par[a])
    return par[a]

def union(a,b):
    global par
    a = find(a)
    b = find(b)
    if a < b:
        par[b] = a
    else:
        par[a] = b


def solution(n, computers):
    global par
    par = [i for i in range(n)]
    l = []
    for i in range(n):
        for j in range(n):
            if computers[i][j] == 1:
                if i != j:
                    union(i,j)
    s = set()
    for i in par:
        s.add(find(par[i]))
    return len(s)

File: 김금환/깊이, 너비 우선 탐색(DFS, BFS)/네트워크.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int p[201];
int answer;

int find(int n){
    if(p[n] < 0) return n;
    return p[n] = find(p[n]);;
}

void merge(int a, int b){
    a = find(a);
    b = find(b);
    if(a == b) return;
    p[b] = a;
    answer--;
}

int solution(int n, vector<vector<int>> computers) {
    fill(p, p + n, -1);
    answer = n;
    
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (computers[i][j] == 1) {
                merge(i, j);
            }
        }
    }
    
    return answer;
}